tfhe/c_api_tests/test_high_level_array.c

```diff
@@ -0,0 +1,123 @@
+#include "tfhe.h"
+
+#include <assert.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <string.h>
+
+// Encrypts a string in a FheUint array
+// No error handling is made, it asserts on all error for demo purposes
+FheUint8 **encrypt_str(const char *const str, const size_t str_len, const ClientKey *ck) {
+  assert(str != NULL && str_len > 0);
+
+  FheUint8 **result = malloc(sizeof(*result) * str_len);
+  assert(result != NULL);
+
+  for (size_t i = 0; i < str_len; ++i) {
+    assert(fhe_uint8_try_encrypt_with_client_key_u8(str[i], ck, &result[i]) == 0);
+  }
+  return result;
+}
+
+void destroy_fhe_uint8_array(FheUint8 **begin, const size_t len) {
+  for (size_t i = 0; i < len; ++i) {
+    fhe_uint8_destroy(begin[i]);
+  }
+  free(begin);
+}
+
+int main(void) {
+  int ok = 0;
+  ConfigBuilder *builder;
+  Config *config;
+
+  config_builder_default(&builder);
+  config_builder_build(builder, &config);
+
+  ClientKey *client_key = NULL;
+  ServerKey *server_key = NULL;
+
+  ok = generate_keys(config, &client_key, &server_key);
+  assert(ok == 0);
+
+  ok = set_server_key(server_key);
+  assert(ok == 0);
+
+  char const *const sentence = "The quick brown fox jumps over the lazy dog";
+  char const *const pattern_1 = "wn fox ";
+  char const *const pattern_2 = "tfhe-rs";
+
+  size_t sentence_len = strlen(sentence);
+  size_t pattern_1_len = strlen(pattern_1);
+  size_t pattern_2_len = strlen(pattern_2);
+
+  assert(pattern_1_len == pattern_2_len); // We use this later in the tests
```

Comment:
where is it used ? I might have missed it

Reply:
This and the other comment `// This one is trivial as the length are not the same` relates to the fact that when calling eq on two slices, if they don't have the same lengths, we trivially know they cant be equal

So one of the test makes sur we go in the path where length ares eq and so we do not trivially know if the slices are equal

Reply:
tbf the comment is not really about that